### DISCUSSION TITLE
Open workspace on currently focused screen

### DISCUSSION THREAD

**Discussion by oars** [i3/i3#5675] [Q&A]
I'm giving i3wm a whirl, but coming from xmonad I am much more used to having workspaces show up on the currently focused screen.

Is there a way to emulate this behaviour in i3wm?

**Comment by orestisfl** [comment]
That's the default behavior, what's in your i3 config?

**Comment by okraits** [comment]
I'm not sure I understood what you mean. Do you mean that

1. new workspaces show up on the currently focused output? That is the default behaviour, as @orestisfl already said.
2. existing workspaces you switch to show up on the currently focused output? i3 binds each workspace to a specific output. Which means that each workspace is displayed on its associated output. To change that behaviour, you need to move the desired workspace to the current output before focusing it. Please have a look at the userguide for hints on how to do that: https://i3wm.org/docs/userguide.html

**Comment by oars** [comment]
Thanks for the comments. Yes I should have specified that my goal is to have this behavior for existing workspaces. I guess the main way to accomplish this would be to use the i3-msg interface to query current outputs and perform the right logic.
